js-packages/@prestojs/viewmodel/src/ViewModelCache.ts

```diff
@@ -167,7 +168,11 @@ function getFieldNameCacheKey(
         } else {
             // getField guarantees that the field will be a RelatedViewModelField so we
             // don't need to check it here
-            const relatedField = viewModel.getField(path.slice(0, -1)) as RelatedViewModelField;
+            const relatedField = viewModel.getField(path.slice(0, -1)) as BaseRelatedViewModelField<
+                any,
+                any,
+                any
+            >;
```

Comment:
Does `any` here really mean any, or does it mean "any, but as long as it satisfies the 'extends' constraints in the class definition"?

Reply:
pretty sure it results in all constraints being ignored. i don't think it really matters here though as don't know anything specific... i guess could specify the constraints again 🤷‍♂️